src/util/listQueryBuilder.js

```diff
@@ -529,57 +529,64 @@ async function performHistoryQuery ({
   const interval = intervals[groupBy]
   const view = groupByViews[groupBy]
 
-  let allResults
-
-  if (useOnlyNonTimeRestrictedFilters && view) {
-    // `AT TIME ZONE 'UTC'` statement is important to add to have the type 'timestamptz'
-    // otherwise the type 'timestamp' is returned and the date won't be UTC
-    let select
-    if (secondaryFilter) {
-      select = knex.raw('?? AT TIME ZONE \'UTC\' as ??, sum(count)::REAL as count', [groupBy, groupBy])
+  const useContinuousAggregate = useOnlyNonTimeRestrictedFilters && view
+
+  const select = (() => {
+    const selectParams = []
+    let selectQuery = ''
+
+    if (useContinuousAggregate) {
+      // Please consult continuous aggregate definition in `migrations/util/timescaleDB.js`
+      // it already embeds `time_bucket` function, thus no need to specify it via request
+      // only reference it via `groupBy` value (hour, day or month)
```

Comment:
Explicit variable `useContinuousAggregate` and this comment are welcome, but the string concatenation makes it hard to see the final result.

Reply:
Right, I'll add the full query for each case.